### ISSUE TITLE
How to use  "ReduceLROnPlateau" in yolov5 training to train custom dataset.

### ISSUE THREAD

**Issue by mabubakarsaleem** [ultralytics/yolov5#9083]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Can,Anyone please help me about the usage of "ReduceLROnPlateau" https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html#torch.optim.lr_scheduler.ReduceLROnPlateau in yolov5 training because i did many changes in the code but got multiple errors to run this.

### Additional

i did changes in the train.py of yolov5 , line# 178 of train.py file of yolov5.

# Scheduler
    if opt.cos_lr:
        lf = one_cycle(1, hyp['lrf'], epochs)  # cosine 1->hyp['lrf']
    else:
        lf = lambda x: (1 - x / epochs) * (1.0 - hyp['lrf']) + hyp['lrf']  # linear
    scheduler = lr_scheduler.LambdaLR(optimizer, lr_lambda=lf)  # plot_lr_scheduler(optimizer, scheduler, epochs)

**Comment by glenn-jocher** [comment]
@mabubakarsaleem you can use the "ReduceLROnPlateau" scheduler in YOLOv5 training by making the following changes in the train.py file:

Replace the existing scheduler code with:
```python
scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, mode='min', factor=0.1, patience=10, verbose=True)
```

This will create a ReduceLROnPlateau scheduler that monitors the specified metric (mode), reduces the learning rate by a factor of 0.1 if no improvement is seen for a certain number of epochs (patience), and provides verbose output. Remember to import the necessary modules at the top of the file.

Hope this helps! You can refer to the Ultralytics Docs for further details on training YOLOv5 models.
